linkd-lib/src/args.rs

```diff
@@ -0,0 +1,298 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+// TODO(xla): Expose discovery args.
+// TODO(xla): Expose storage args.
+// TODO(xla): Expose logging args.
+
+use std::{fmt, net::SocketAddr, path::PathBuf, str::FromStr};
+
+use structopt::StructOpt;
+
+use librad::{
+    crypto,
+    net::Network,
+    profile::{ProfileId, RadHome},
+    PeerId,
+};
+
+#[derive(Debug, Default, Eq, PartialEq, StructOpt)]
+pub struct Args {
+    /// List of bootstrap nodes for initial discovery.
+    #[structopt(long = "bootstrap", name = "bootstrap")]
+    pub bootstraps: Vec<Bootstrap>,
+
+    /// Identifier of the profile the daemon will run for. This value determines
+    /// which monorepo (if existing) on disk will be the backing storage.
+    #[structopt(long)]
+    pub profile_id: Option<ProfileId>,
+
+    /// Home of the profile data, if not provided is read from the environment
+    /// and falls back to project dirs.
+    #[structopt(long, default_value, parse(from_str = parse_rad_home))]
+    pub rad_home: RadHome,
+
+    /// Configures the type of signer used to get access to the storage.
+    #[structopt(long, default_value)]
+    pub signer: Signer,
+
+    #[structopt(flatten)]
+    pub key: KeyArgs,
+
+    #[structopt(flatten)]
+    pub metrics: MetricsArgs,
+
+    #[structopt(flatten)]
+    pub protocol: ProtocolArgs,
+
+    /// Forces the creation of a temporary root for the local state, sould be
+    /// used for debug and testing only.
+    #[structopt(long)]
+    pub tmp_root: bool,
```

Comment:
IIUC, setting this to `true` will render `rad_home` a no-op. Perhaps fold into one argument?